luigi/contrib/batch.py

```diff
@@ -0,0 +1,213 @@
+.. _awscli: https://aws.amazon.com/cli
+.. _`AWS Batch User Guide`: http://docs.aws.amazon.com/AmazonECS/latest/developerguide/ECS_GetStarted.html
+
+"""
+
+import json
+import logging
+import random
+import string
+import time
+
+import luigi
+logger = logging.getLogger(__name__)
+
+try:
+    import boto3
+except ImportError:
+    logger.warning('boto3 is not installed. BatchTasks require boto3')
+
+
+class BatchJobException(Exception):
+    pass
+
+
+POLL_TIME = 10
+
+
+def _random_id():
+    return 'batch-job-' + ''.join(random.sample(string.ascii_lowercase, 8))
+
+
+class BatchClient(object):
+
+    def __init__(self):
+        self._client = boto3.client('batch')
+        self._log_client = boto3.client('logs')
+        self._queue = self.get_active_queue()
+    
+    def get_active_queue(self):
+        """Get name of first active job queue"""
+        
+        # Get dict of active queues keyed by name
+        queues = {q['jobQueueName']:q for q in self._client.describe_job_queues()['jobQueues']
+                  if q['state'] == 'ENABLED' and q['status'] == 'VALID'}
+        if not queues:
+            raise Exception('No job queues with state=ENABLED and status=VALID')
+
+        # Pick the first queue as default
+        return list(queues.keys())[0]
+
+    def get_job_id_from_name(self, job_name):
+        """Retrieve the first job ID matching the given name"""
+        jobs = self._client.list_jobs(jobQueue=self._queue, jobStatus='RUNNING')['jobSummaryList']
+        matching_jobs = [job for job in jobs if job['jobName'] == job_name]
+        if matching_jobs:
+            return matching_jobs[0]['jobId']
+
+    def get_job_status(self, job_id):
+        """Retrieve task statuses from ECS API
+
+        Args:
+            job_id (str): AWS Batch job uuid
+
+        Returns list of {SUBMITTED|PENDING|RUNNABLE|STARTING|RUNNING|SUCCEEDED|FAILED} for each id in job_ids
+        """
+        response = self._client.describe_jobs(jobs=[job_id])
+
+        # Error checking
+        status_code = response['ResponseMetadata']['HTTPStatusCode']
+        if status_code != 200:
+            msg = 'Job status request received status code {0}:\n{1}'
+            raise Exception(msg.format(status_code, response))
+
+        return response['jobs'][0]['status']
+
+    def get_logs(self, log_stream_name, get_last=50):
+        """Retrieve log stream from CloudWatch"""
+        response = self._log_client.get_log_events(
+            logGroupName='/aws/batch/job', 
+            logStreamName=log_stream_name,
+            startFromHead=False)
+        events = response['events']
+        return '\n'.join(e['message'] for e in events[-get_last:])
+
+    def submit_job(self, job_definition, parameters, job_name=None, queue=None):
+        """Wrap submit_job with useful defaults"""
+        if job_name is None:
+            job_name = _random_id()
+        response = self._client.submit_job(
+            jobName = job_name,
```

Comment:
Remove the explicit conditional check for if `job_name is None` by updating this to `jobName=job_name or _random_id(),`